src/renderer/api/kube-watch-api.ts

```diff
@@ -274,12 +275,29 @@ export class KubeWatchApi {
       try {
         const kubeEvent: IKubeWatchEvent = JSON.parse(json);
         const message = this.getMessage(kubeEvent);
+        const { data, namespace } = message;
 
-        if (!this.namespaces.includes(message.namespace)) {
-          continue; // skip updates from non-watching resources context
+        if (!this.isAllowedApi(message.api)) {
+          return;
```

Comment:
`return "";` ?

I find this method confusing. If this event is for an api that is not allowed then no message is emitted for it and all remaining events in `events` are abandoned? Similarly, if there is an error, it's caught below and this event is returned but again all remaining events in `events` are abandoned?

Reply:
Yes, you're right. Probably should be remove or replaced with `continue`.

Reply:
Removed.